# Special Keys
bool k_up         	# up arrow
bool k_down        	# down arrow
bool k_right       	# right arrow
bool k_left         # left arrow	
bool k_backspace	# backspace
bool k_tab			# tab
bool k_clear		# clear 
bool k_return		# return
bool k_pause		# pause
bool k_escape		# escape
bool k_space  		# space
bool k_insert       # insert
bool k_home         # home
bool k_end          # end
bool k_pageup       # page up
bool k_pagedown     # page down
bool k_numlock      # numlock
bool k_capslock     # capslock
bool k_scrollock    # scrollock
bool k_rshift       # right shift
bool k_lshift       # left shift
bool k_rctrl        # right ctrl
bool k_lctrl        # left ctrl
bool k_ralt         # right alt
bool k_lalt         # left alt
bool k_rmeta        # right meta
bool k_lmeta        # left meta
bool k_lsuper       # left windows key
bool k_rsuper       # right windows key
bool k_mode         # mode shift
bool k_help         # help
bool k_print        # print screen
bool k_sysreq       # sysrq
bool k_break        # break
bool k_menu         # menu
bool k_power        # power
bool k_euro         # euro

# Numbers
bool k_0            # 0
bool k_1            # 1
bool k_2           	# 2
bool k_3           	# 3
bool k_4            # 4
bool k_5          	# 5
bool k_6            # 6
bool k_7           	# 7
bool k_8          	# 8
bool k_9        	# 9

# Punctuation
bool k_colon       	# colon
bool k_semicolon  	# semicolon
bool k_less       	# less-than sign
bool k_equals      	# equals sign
bool k_greater     	# greater-than sign
bool k_question   	# question mark
bool k_at          	# at
bool k_leftbracket 	# left bracket
bool k_backslash   	# backslash
bool k_rightbracket # right bracket
bool k_caret       	# caret
bool k_underscore   # underscore
bool k_backquote   	# grave
bool k_exclaim     	# exclaim 
bool k_quotedbl     # quotedbl
bool k_hash        	# hash
bool k_dollar      	# dollar
bool k_ampersand    # ampersand
bool k_quote        # quote
bool k_leftparen    # left parenthesis
bool k_rightparen   # right parenthesis
bool k_asterisk     # asterisk
bool k_plus         # plus sign
bool k_comma        # comma
bool k_minus        # minus sign
bool k_period       # period
bool k_slash        # forward slash

# Letters
bool k_a            # a
bool k_b           	# b
bool k_c           	# c
bool k_d           	# d
bool k_e           	# e
bool k_f           	# f
bool k_g           	# g
bool k_h           	# h
bool k_i           	# i
bool k_j           	# j
bool k_k           	# k
bool k_l           	# l
bool k_m           	# m
bool k_n         	# n
bool k_o           	# o
bool k_p           	# p
bool k_q           	# q 
bool k_r           	# r
bool k_s           	# s 
bool k_t           	# t 
bool k_u           	# u 
bool k_v           	# v
bool k_w           	# w
bool k_x           	# x
bool k_y           	# y
bool k_z           	# z

# Keypad
bool k_delete     	# delete
bool k_kp0         	# keypad 0
bool k_kp1         	# keypad 1
bool k_kp2        	# keypad 2
bool k_kp3         	# keypad 3
bool k_kp4      	# keypad 4
bool k_kp5			# keypad 5
bool k_kp6          # keypad 6
bool k_kp7          # keypad 7
bool k_kp8          # keypad 8
bool k_kp9          # keypad 9
bool k_kp_period   	# keypad period
bool k_kp_divide   	# keypad divide
bool k_kp_multiply 	# keypad multiply
bool k_kp_minus    	# keypad minus
bool k_kp_plus     	# keypad plus
bool k_kp_enter    	# keypad enter
bool k_kp_equals   	# keypad equals

# F keys
bool k_f1          	# f1
bool k_f2           # f2
bool k_f3         	# f3
bool k_f4           # f4
bool k_f5           # f5
bool k_f6          	# f6
bool k_f7           # f7
bool k_f8           # f8
bool k_f9           # f9
bool k_f10          # f10
bool k_f11          # f11
bool k_f12          # f12
bool k_f13          # f13
bool k_f14          # f14
bool k_f15          # f15
